lib/app.js

```diff
@@ -0,0 +1,43 @@
+const { createServer } = require('net');
+const ChatRoom = require('./ChatRoom');
+const parseMessage = require('./parseMessage');
+
+const server = createServer(client => {
+  const chatroom = new ChatRoom();
+  client.setEncoding('utf8');
+
+  server.on('error', err => {
+    console.log(err);
+  });
+
+
+  client.on('data', data => {
+    console.log(`${client.username} connected`);
+    // add client to chatroom
+    chatroom.add(client);
```

Comment:
```suggestion
    chatroom.add(client);
```
Try moving this to line 8 and see what happens...